framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2243,6 +2248,14 @@ public Void visitAnnotation(AnnotationTree tree, Void p) {
       return null;
     }
 
+    if (this.isPreOrPostConditionAnnotation(annoName) && methodTree != null) {
+      ExecutableElement methodElement = TreeUtils.elementFromDeclaration(methodTree);
+      Set<Contract> contracts = atypeFactory.getContractsFromMethod().getContracts(methodElement);
```

Comment:
@smillst To me, this feels like a very roundabout way to get the annotations for a method (specifically, the contract annotations like `@EnsuresQualifier` or `@EnsuresQualifierIf`) when we're already visiting the annotation itself; do you know of a better API that lets me get this directly without having to go through the `methodTree`?